addons/html_editor/static/src/main/align/align_plugin.js

```diff
@@ -0,0 +1,123 @@
+import { Plugin } from "@html_editor/plugin";
+import { closestBlock } from "@html_editor/utils/blocks";
+import { isVisibleTextNode } from "@html_editor/utils/dom_info";
+import { withSequence } from "@html_editor/utils/resource";
+import { _t } from "@web/core/l10n/translation";
+import { AlignSelector } from "./align_selector";
+import { reactive } from "@odoo/owl";
+
+const alignmentItems = [
+    { mode: "left" },
+    { mode: "center" },
+    { mode: "right" },
+    { mode: "justify" },
+];
+
+function hasBlockTextAlignment(nodes) {
+    for (const node of nodes) {
+        const block = closestBlock(node);
+        if (block && block.style["text-align"]) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
Considering the previous comment about predicates taking only one node, this becomes:
```js
node => closestBlock(node)?.style.textAlign;
```
So I suggest to inline it at call site since it's only used once and is very short.